### ISSUE TITLE
[Bug]: Git sync modal shows uncommitted changes after committing and pushing changes to git.

### ISSUE THREAD

**Issue by Parthvi12** [appsmithorg/appsmith#23070]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

https://user-images.githubusercontent.com/80334441/236770355-73fc2234-8b3a-4a81-8d3b-000dbdc4733c.mov

### Steps To Reproduce

1. Fork template 'Maintenance Order Management'
2. Connect app to git
3. Observe  uncommitted changes are shown.

### Environment

Release

### Version

Release

**Comment by AnaghHegde** [comment]
@Parthvi12 Can you share the diff? What changes are committed to the repo?

**Comment by Parthvi12** [comment]
@AnaghHegde : https://github.com/Parthvi12/test-123/commit/d8fcbcde468c3d105567c113e286bf1dbbc29c5e and this is initial commit: https://github.com/Parthvi12/test-123/commit/3ef2a2e0c66521b3d27308f6eaaa8554628180de

**Comment by AnaghHegde** [comment]
The datasource was deleted from the repo. Can you reproduce this here?

**Comment by Parthvi12** [comment]
@AnaghHegde Yes on committing changes the datasource gets deleted but I didn't delete it from app as you can see in recording, I am able to reproduce this issue after connecting 'Maintenance Order Management' template to git.

**Comment by infinitetrooper** [comment]
@Parthvi12  - is this still reproducable?

**Comment by RakshaKShetty** [comment]
@infinitetrooper This issue is still reproducible

**Comment by nidhi-nair** [comment]
While the maintenance order management template needs to be cleaned up to not require an extra commit, on the git front, we need to check why this is requiring an extra commit from the user as well. However, considering that this would mostly only impact templates/blocks, lowering the priority and disassociating from clean up.
